drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/handler/EdgeDriverBinaryHandler.java

```diff
@@ -55,80 +56,64 @@ protected String getUrlToDownloadProperty() {
 
     @Override
     protected ExternalBinarySource getExternalBinarySource() {
-        return new EdgeStorageSources();
+        return new EdgeDriverBinaryHandler.EdgeStorageSources("https://msedgedriver.azureedge.net/");
     }
 
     @Override
     protected DesiredCapabilities getCapabilities() {
         return capabilities;
     }
 
-    private class EdgeStorageSources implements ExternalBinarySource {
+    private class EdgeStorageSources extends XmlStorageSource {
 
-        private static final String EDGE_WEB_DRIVERS_URL =
-            "https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/";
-        private static final String DRIVERS_LIST = ".driver-downloads li";
-        private static final String INFO_PARAGRAPH = "p";
-        private static final String LINK = "a";
-        private static final String URL = "href";
-        private static final int VERSION_POSITION = 1;
-        private static final int LATEST_DRIVER = 0;
-        private Logger log = Logger.getLogger(EdgeStorageSources.class.toString());
-
-        @Override
-        public ExternalBinary getLatestRelease() throws Exception {
-            Element driver = getDriversList().get(LATEST_DRIVER);
-            String driverVersion = getDriverVersion(driver);
-            String webDriverUrl = getDriverUrl(driver);
-
-            return new ExternalBinary(driverVersion, webDriverUrl);
+        EdgeStorageSources(String baseUrl) {
+            super("Blob", "Name", baseUrl, "https://msedgewebdriverstorage.blob.core.windows.net/edgewebdriver/LATEST_STABLE", new HttpClient());
         }
 
-        @Override
-        public ExternalBinary getReleaseForVersion(String version) throws Exception {
-
-            String webDriverUrl = "";
-            String driverVersion = "";
-
-            Elements driversList = getDriversList();
-
-            for (int driverItemNumber = 0; driverItemNumber < driversList.size(); driverItemNumber++) {
-                Element driver = driversList.get(driverItemNumber);
-                driverVersion = getDriverVersion(driver);
-
-                if (driverVersion.equals(version)) {
-                    webDriverUrl = getDriverUrl(driver);
-                    break;
-                }
-            }
-
-            if (webDriverUrl.isEmpty()) {
-                log.warning("WebDriver with specified version was not found.");
-            }
+        protected String getExpectedKeyRegex(String requiredVersion, String directory) {
+            return Pattern.quote(requiredVersion + "/" + getFileNameRegexToDownload(requiredVersion));
+        }
```

Comment:
Same as the other comment, it uses only the first parameter

Reply:
See https://github.com/arquillian/arquillian-extension-drone/pull/288#discussion_r741930672